### PR TITLE
Add ability to specify personaId and contentChannelIds for Persona Feeds

### PR DESCRIPTION
**Pull request by conrad-vanl**
### What does this PR do, or why is it needed?

This is something I've been using on Willow to allow customization of the `personaId` and `contentChannelIds` selected for a `personaFeed`. 

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by conrad-vanl** [comment]
@vinnyjth looking for thoughts on this before I add tests and such.

The only thing that I could see this really needing that this doesn't address is instead of specifying a single persona to select, specifying a category of personas. Not sure if that'd be easy to do instead.

File: packages/apollos-data-connector-rock/src/content-items/data-source.js [hunk 1]
```diff
@@ -339,36 +339,40 @@ export default class ContentItem extends RockApolloDataSource {
...
-    // Use plugin, if the user has set USE_PLUGIN to true.
-    // In general, you should ALWAYS use the plugin if possible.
-    const endpoint = get(ApollosConfig, 'ROCK.USE_PLUGIN', false)
-      ? 'Apollos/ContentChannelItemsByDataViewGuids'
-      : 'ContentChannelItems/GetFromPersonDataView';
-
     // Grabs content items based on personas
-    return this.request(
-      `${endpoint}?guids=${getPersonaGuidsForUser
+    let request = this.request(
```
Comment (vinnyjth) [inline packages/apollos-data-connector-rock/src/content-items/data-source.js]: Make sure you don't ditch the code that uses the plugin endpoint, I think you need to revert your changes to line 360 and 361 :)

File: packages/apollos-data-connector-rock/src/content-items/data-source.js [hunk 2]
```diff
@@ -339,36 +339,40 @@ export default class ContentItem extends RockApolloDataSource {
...
+    contentChannelIds,
+  } = {}) => {
     const {
       dataSources: { Person },
     } = this.context;
 
     // Grabs the guids associated with all dataviews user is memeber
     const getPersonaGuidsForUser = await Person.getPersonas({
-      categoryId: ROCK_MAPPINGS.DATAVIEW_CATEGORIES.PersonaId,
+      categoryId: personaId,
```
Comment (vinnyjth) [inline packages/apollos-data-connector-rock/src/content-items/data-source.js]: Yah, we'll have to augment this function to support fetching data from more than one `personaId`. Not too bad as long as we don't mind making additional queries... I'm fine leaving it out for now.

File: packages/apollos-data-connector-rock/src/content-items/data-source.js [hunk 3]
```diff
@@ -339,36 +339,40 @@ export default class ContentItem extends RockApolloDataSource {
...
+    let request = this.request(
+      `ContentChannelItems/GetFromPersonDataView?guids=${getPersonaGuidsForUser
         .map((obj) => obj.guid)
         .join()}`
     )
       .andFilter(this.LIVE_CONTENT())
-      .top(first)
       .orderBy('StartDateTime', 'desc');
+
+    if (contentChannelIds && contentChannelIds.length) {
```
Comment (vinnyjth) [inline packages/apollos-data-connector-rock/src/content-items/data-source.js]: 👍

File: packages/apollos-data-connector-rock/src/content-items/resolver.js [hunk 4]
```diff
@@ -70,9 +70,7 @@ const resolver = {
         args,
       }),
     personaFeed: async (root, args, { dataSources }) => {
-      const personaFeed = await dataSources.ContentItem.byPersonaFeed(
-        args.first
-      );
+      const personaFeed = await dataSources.ContentItem.byPersonaFeed(args);
```
Comment (vinnyjth) [inline packages/apollos-data-connector-rock/src/content-items/resolver.js]: 👍

File: packages/apollos-data-connector-rock/src/features/data-source.js [hunk 5]
```diff
@@ -63,11 +63,15 @@ export default class Features extends RockApolloDataSource {
     };
   }
 
-  async personaFeedAlgorithm() {
+  async personaFeedAlgorithm({ personaId, contentChannelIds, first = 3 }) {
```
Comment (vinnyjth) [inline packages/apollos-data-connector-rock/src/features/data-source.js]: I like doing this, but I think we need to make sure that we either have defaults here or in the `config.yml`. I think defaults in the arguments should be fine.

Comment (vinnyjth) [reply packages/apollos-data-connector-rock/src/features/data-source.js]: I know there are defaults inside the `byPersonaFeed` method, so I guess we don't _need_ them :)

**Comment by conrad-vanl** [comment]
@vinnyjth I never went back and worked on this more.

It would probably be a _great_ idea to try and bring in some of the feature expansion we did on Willow into core this cycle. What would you think?

**Comment by vinnyjth** [comment]
> It would probably be a great idea to try and bring in some of the feature expansion we did on Willow into core this cycle. What would you think?

Totally agree. Do we need to meet and figure out how (and what) we need to focus on? There's a bit here, so I think we'll need to add it to basecamp.

**Pull request closed without merging**